tasks/eva_2889/remediate_duplicate_remapped_ss/src/RemediateDuplicateRemappedSS.groovy

```diff
@@ -0,0 +1,399 @@
+import static org.springframework.data.mongodb.core.query.Criteria.where
+
+
+@Component
+@Import(value=[SSSplitWriterConfiguration.class,
+        InputParametersConfiguration.class, MetricConfiguration.class, ListenersConfiguration.class,
+        SubmittedVariantAccessioningConfiguration.class])
+class RemediateDuplicateRemappedSS implements CommandLineRunner {
+
+    private static final Logger logger = LoggerFactory.getLogger(RemediateDuplicateRemappedSS.class)
+
+    @Autowired
+    private SubmittedVariantAccessioningService submittedVariantAccessioningService
+
+    @Autowired
+    private MongoTemplate destinationMongoTemplate
+
+    private MongoTemplate devMongoTemplate
+
+    @Value('${devenv.properties}')
+    private String devEnvPropertiesFile
+
+    @Value('${assembly.to.remediate}')
+    private String assemblyToRemediate
+
+    @Autowired
+    private Long accessioningMonotonicInitSs
+
+    private Criteria criteriaToFindAssembly
+
+    private Criteria criteriaToFindRemappedEntries
+
+    private String remappedAttributesCollection = "remappedAttributes"
+
+    private SSDeprecationWriter ssDeprecationWriter
+
+    void run(String... args) {
+        this.criteriaToFindAssembly = where("seq").is(assemblyToRemediate)
+        this.criteriaToFindRemappedEntries = where("remappedFrom").exists(true)
+        this.devMongoTemplate = EVADatabaseEnvironment.parseFrom(devEnvPropertiesFile).mongoTemplate
+        this.devMongoTemplate.setWriteConcern(WriteConcern.MAJORITY)
+        this.ssDeprecationWriter = new SSDeprecationWriter(this.destinationMongoTemplate, this.accessioningMonotonicInitSs)
+
+        remediateDuplicateRemappedSSCollection(DbsnpSubmittedVariantEntity.class)
+        remediateDuplicateRemappedSSCollection(SubmittedVariantEntity.class)
+    }
+
+    void remediateDuplicateRemappedSSCollection(Class<? extends SubmittedVariantEntity> collectionClassToUse) {
+        int numSVEScanned = 0
+        int batchIndex = 0
+        String lastSeenID = null
+        while(true) {
+            ImmutablePair<List<? extends SubmittedVariantEntity>, String> svesAndLastSeenID =
+                    getNextBatchOfDuplicateRemappedSVEs(lastSeenID, collectionClassToUse)
+            if (svesAndLastSeenID != null) {
+                remediateDuplicateRemappedSSNotGeneratedByRemapping(svesAndLastSeenID.left, collectionClassToUse)
+                numSVEScanned += svesAndLastSeenID.left.size()
+                lastSeenID = svesAndLastSeenID.right
+                logger.info("Processed " + numSVEScanned + " remapped IDs in ${collectionClassToUse.getTypeName()}...")
+                batchIndex += 1
+            } else {
+                break
+            }
+        }
+    }
+
+    @Retryable(value = MongoCursorNotFoundException.class, maxAttempts = 5, backoff = @Backoff(delay = 100L))
+    ImmutablePair<List<? extends SubmittedVariantEntity>, String> getNextBatchOfDuplicateRemappedSVEs(
```

Comment:
Since we have a very specific [set of accessions to deprecate](https://www.ebi.ac.uk/panda/jira/browse/EVA-2889?focusedCommentId=396346&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-396346), we are using the entire SVE/dbsnpSVE collections in DEV as the source.

Reply:
Makes sense